Outreach: `.scss` refactoring

nmajor25 opened the pull request:
Shortcut Story ID: [sc-39249]

nmajor25 committed:
Delete un-used styles/class names

nmajor25 committed:
Consolidate icon styles to dialog.scss file and 'dialog__icon dialog__icon--' class names

nmajor25 committed:
Remove un-needed instances of 'text-align: left'

In #1139 we removed '.app-root { text-align: center; }', so we no longer need the left alignment styles to counteract that

nmajor25 committed:
On '.dialog', switch default font size from 24px => 20px

There were a few error view with font-size: 24px, but should be font-size: 20px instead

nmajor25 commented on src/js/outreach/views/opt-in.scss:
```diff
@@ -44,12 +44,6 @@
-    border-color: $error-color;
-    color: $error-color;
-    margin-bottom: 6px;
-  }
```
We're only doing error styling for inputs on the verify code view. The opt in view doesn't show any errors for input fields. So this styling never gets used.

nmajor25 commented on src/js/outreach/views/dialog.scss:
```diff
@@ -10,10 +10,6 @@
 
-.icon {
-  font-size: 32px;
-}
```
Only thing using this was the form submit success icon. But it's no longer being used after that was changed in #1138.

All icons are now using `.dialog__icon > .icon`.

paulfalgout commented on src/js/outreach/views/dialog.scss:
```diff
@@ -10,104 +10,38 @@
-    border-color: $error-color;
-    color: $error-color;
-    margin-bottom: 6px;
+  &.dialog__icon--success {
```
I don't think the `&`s are necessary here.  these classes look to be BEM'd enough that they do not need to be nested.  They just need to be either after the `dialog_icon` in the cascade or since it's all in the same file these could extend the base such that only this class is necessary in the DOM.  Ideally `&` is saved for adding states generally prefixed by something like `&.has-`, `&.is-` etc

nmajor25 replied:
I went the `@extend` route. Works well. I like that we now only need the `.dialog__icon--` classname in the dom. Instead of needing `.dialog__icon .dialog__icon--`. Good call 👍

nmajor25 committed:
Extend '.dialog__icon--' styles from '.dialog__icon'

This allows us to only have the '.dialog__icon--' classname in the dom, instead of needing '.dialog__icon .dialog__icon--' for each icon
